pkg/daemon/update.go

```diff
@@ -1918,12 +1923,31 @@ func (dn *Daemon) reboot(rationale string) error {
 	return nil
 }
 
-func (dn *CoreOSDaemon) applyLayeredOSChanges(mcDiff machineConfigDiff, oldConfig, newConfig *mcfgv1.MachineConfig) (retErr error) {
+// changeBootedImage takes an override and updates the current config on disk to use this osImageURL
+func (dn *CoreOSDaemon) changeBootedImage(override string) error {
+	oldMC, err := dn.getCurrentConfigOnDisk()
+	if err != nil {
+		return err
+	}
+	newMC := oldMC.DeepCopy()
+	newMC.Spec.OSImageURL = override
+
+	return dn.storeCurrentConfigOnDisk(newMC)
+
+}
+
+func (dn *CoreOSDaemon) applyLayeredOSChanges(mcDiff machineConfigDiff, oldConfig, newConfig *mcfgv1.MachineConfig, force bool) (retErr error) {
 	// Override the computed diff if the booted state differs from the oldConfig
 	// https://issues.redhat.com/browse/OCPBUGS-2757
 	if mcDiff.osUpdate && dn.bootedOSImageURL == newConfig.Spec.OSImageURL {
-		klog.Infof("Already in desired image %s", newConfig.Spec.OSImageURL)
-		mcDiff.osUpdate = false
+		if force {
+			if err := dn.changeBootedImage(""); err != nil {
```

Comment:
I don't think this is necessary if we're forcing it through to here. I think it's also safe to remove the mcDiff.osUpdate check from above and continue with the rest via force. Even if we force, I think if dn.bootedOSImageURL == newConfig.Spec.OSImageURL means that we are on the right version